packages/mui-material/src/Popover/Popover.js

```diff
@@ -87,8 +87,8 @@ const PopoverPaper = styled(Paper, {
   // It's most likely on issue on userland.
   minWidth: 16,
   minHeight: 16,
-  maxWidth: 'calc(100% - 32px)',
-  maxHeight: 'calc(100% - 32px)',
+  maxWidth: 'calc(100% - ${marginThreshold})',
+  maxHeight: 'calc(100% - ${marginThreshold})',
```

Comment:
I didn't look at the problem, so no idea if the change is correct, but why change the value?

```suggestion
  maxWidth: 'calc(100% - ${marginThreshold * 2})',
  maxHeight: 'calc(100% - ${marginThreshold * 2})',
```

Reply:
Currently the value is hard-coded for the default threshold of 16px

If anything other than that is specified as a margin threshold, the popover becomes taller than the screen

The original issue explains it better

Also the current fix is probably incorrect, I was just giving a pseudo code example of a fix.i can take a look later today and help Nour with the resolution